src/esp/sim/Simulator.cpp

```diff
@@ -916,6 +916,89 @@ assets::MeshData::ptr Simulator::getJoinedMesh(
   return joinedMesh;
 }
 
+std::vector<vec3f> Simulator::getStageMeshVerts() {
+  assets::MeshData::ptr joinedMesh = getJoinedMesh(true);
+  return joinedMesh->vbo;
+}
+
+std::vector<vec3ui> Simulator::getStageMeshFaceIdxs() {
+  assets::MeshData::ptr joinedMesh = getJoinedMesh(true);
```

Comment:
This part is fine for now, but will be an efficiency problem. You'll also be re-extracting the full mesh each time. Better to break it up and get a point-cloud per-object. Then you can dynamically update only the moving parts of the cloud.